apps/docs/src/guide/contracts/multicalls.md

```diff
@@ -1,25 +1,21 @@
-# Multiple contract calls
+# Multiple Contract Calls
 
-At a basic level, you can call the same contract function multiple times in the same transaction:
+You can execute multiple contract calls in a single transaction, either to the same contract or to different contracts. This can improve efficiency and reduce the overall transaction costs.
 
-<<< @/../../../packages/fuel-gauge/src/contract.test.ts#Contract-multicall{ts:line-numbers}
+## Calling the Same Contract Functions Multiple Times
 
-## Different contracts in multicalls
-
-You can execute multiple contract calls to distinct contracts within a single transaction. To achieve this, you first prepare all the contract calls that you want to bundle:
+Use the `multiCall` method to call the same contract functions multiple times in a single transaction:
 
-<<< @/../../../packages/fuel-gauge/src/contract.test.ts#Contract-multicall-multiple-contracts{ts:line-numbers}
+<<< @/../../docs-snippets/src/guide/contracts/multicalls.test.ts#multicall-1{ts:line-numbers}
 
-You can also set call parameters, variable outputs, or external contracts for every contract call, as long as you don't execute it with `call()` or `simulate()`. See also [call parameters](./call-parameters.md).
-
-Next, you provide the prepared calls to the `multiCall` method and optionally configure transaction parameters:
+## Different contracts in multicalls
 
-<<< @/../../../packages/fuel-gauge/src/contract.test.ts#Contract-multicall-multiple-contracts-p2{ts:line-numbers}
+The `multiCall` method also allows you to execute multiple contract calls to distinct contracts within a single transaction:
 
-> **Note:** any transaction parameters configured on separate contract calls are disregarded in favor of the parameters provided to `multiCall`.
+<<< @/../../docs-snippets/src/guide/contracts/multicalls.test.ts#multicall-2{ts:line-numbers}
 
-## Output values
+You can also chain supported contract call methods, like `callParams`, for each contract call:
 
-To get the output values of the bundled calls, invoke of `call()` or `simulate()`:
+<<< @/../../docs-snippets/src/guide/contracts/multicalls.test.ts#multicall-3{ts:line-numbers}
 
-<<< @/../../../packages/fuel-gauge/src/contract.test.ts#Contract-multicall-multiple-contracts-p3{ts:line-numbers}
+> **Note:** When chaining contract call methods within `multiCall`, avoid executing the contract functions themselves, such as `.call`, `.get`, and `.simulate`.
```

Comment:
Do we know the potential harm that can be caused by this? If yes, maybe it would be a helpful piece of small information that we can add here.